src/cdk/scrolling/virtual-scroll-viewport.ts

```diff
@@ -202,6 +204,17 @@ export class CdkVirtualScrollViewport extends CdkVirtualScrollable implements On
       this.elementRef.nativeElement.classList.add('cdk-virtual-scrollable');
       this.scrollable = this;
     }
+
+    const ref = effect(
+      () => {
+        if (!this._changeDetectionNeeded()) {
+          return;
```

Comment:
nit: maybe invert this and remove the `return` since there's only one statement after it.